server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/service/ExportService.java

```diff
@@ -83,6 +84,14 @@ public ExportService(MailboxBackup mailboxBackup, BlobStore blobStore, BlobExpor
                 .export()));
     }
 
+    private void deleteBlob(BlobId blobId) {
+        blobStore.delete(blobStore.getDefaultBucketName(), blobId)
+            .onErrorResume(e -> {
+                LOGGER.error("Error deleting Blob with blobId: {}", blobId.asString(), e);
+                return Mono.empty();
+            }).block();
```

Comment:
Can't we find a way where we don't need to block this operation?

Reply:
we can use subscribe() here instead. Would that be ok?

Reply:
`subscribe` here is not better than `block`. you have to flatMap that Mono

Reply:
The thing is this block of code is being passed to the doFinally operator as a Consumer, so in order for it to be executed i have to subscribe to the Mono inside it by using either block() or subscribe(), otherwise nothing will happen when we arrive at the doFinally block?